be/src/olap/rowset/segment_v2/column_reader.cpp

```diff
@@ -137,6 +137,7 @@ Status ColumnReader::create(const ColumnReaderOptions& opts, const ColumnMetaPB&
         }
         case FieldType::OLAP_FIELD_TYPE_MAP: {
             // map reader now has 3 sub readers for key, value, offsets(scalar), null(scala)
+            DCHECK(meta.children_columns_size() == 3 || meta.children_columns_size() == 4);
```

Comment:
why not only 3?

Reply:
if map column is nullable, meta.children_columns(3) means to read null map